qTox/qTox
# Add audio backend that supports echo canceling

This PR adds a second audio backend (still using OpenAL) that can provide echo canceling. I don't think think this can be merged right away, but I think it's now in a state where testing can begin.

Working State:
- Echo canceling working on my laptop (Thinkpad T440p, Pulseaudio). I tested this with 2 qTox clients Alice and Bob. Alice was set to capture from the laptop mic and play back over the laptop speakers. Bob was set to capture from a virtual pulseaudio input streaming some music (any independent input will work) and playback over a headset. This way, Bob is emulating a Friend and Alice is yourself. Now without echo cancellation enabled, the music was clearly audible as noise in the headset, when enabling echo cancellation it wasn't. ✔️ 
- fallback if OpenAL extensions needed are missing implemented, but untested.

Not Working:

- echobot, It seems echobot creates an echo that is to strong to filter out using libfilteraudio, IDK what to do about this.
- stereo, libfilteraudio only supports mono audio, I don't think it will be a problem to limit echo canceling to mono since it's not suitable for high quality streaming anyway.

To Test:

- different systems, we need to test this on as many computers as possible, because they all have different audio latencies which are a critical part of echo canceling.
- Windows, theoretically this should also work on Windows out of the box, but it's not tested.

To Do:
- add libfilteraudio dependencies to documents and build system
- improve switching between the two audio backends


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/4367)
